src/main/java/org/springframework/samples/petclinic/controllers/GameController.java

```diff
@@ -231,4 +258,41 @@ public ResponseEntity<List<Game>> getAll() {
 
         return new ResponseEntity<List<Game>>(gameList.get(), HttpStatus.OK);
     }
+
+
+
+    private void initializeGame(Game currentGame) {
+        try {
+            Integer firstIndex = 0;
+            Integer secondIndex = 4;
+            List<Card> allCards = cardService.findAll().orElseThrow(() -> new RuntimeException("No se encontraron cartas."));
+
+            Collections.shuffle(allCards);
+
+            currentGame.setCentralCard(allCards.get(17));
+
+            List<GamePlayer> gamePlayers = new ArrayList<>(currentGame.getRaw_game_players());
+            for (GamePlayer gamePlayer : gamePlayers) {
+                // De momento repartimos 3 cartas a cada jugador
+
+                List<Card> playerCards = new ArrayList<>(allCards.subList(firstIndex, secondIndex));
+                Hand playerHand = new Hand();
+                playerHand.setCards(playerCards);
+                handService.saveHand(playerHand);
+                gamePlayer.setHand(playerHand);
+                gamePlayerService.saveGamePlayer(gamePlayer);
+                
+                firstIndex += 4;
+                secondIndex += 4;
+                
+            }
+            currentGame.setStart(LocalDateTime.now());
+        } catch (RuntimeException e) {
+            System.err.println("Error al inicializar el juego: " + e.getMessage());
+            e.printStackTrace();
+        } catch (Exception e) {
+            System.err.println("Error general al inicializar el juego.");
+            e.printStackTrace();
+        }
+    }
```

Comment:
Esto debe de estar en un servicio, como vean esto aquí los profesores nos crujen jajaja

Reply:
Cierto jajajaj

Reply:
Pero puedo llamar a un servicio dentro de otro servicio? Es que lo estoy intentando poner el método en el GameService, pero me pone esto: Action:

Relying upon circular references is discouraged and they are prohibited by default. Update your application to remove the dependency cycle between beans. As a last resort, it may be possible to break the cycle automatically by setting spring.main.allow-circular-references to true.

Reply:
Claro, ese error creo que es porque dentro de GameService estás importando otro servicio que a su vez importa de nuevo a GameService.

Reply:
Pues no se cómo hacerlo sin que me de el error este